ui/app/adapters/secret-engine.js

```diff
@@ -42,7 +42,7 @@ export default ApplicationAdapter.extend({
       }
     } catch (error) {
       // no path means this was an error on listing
-      if (!query.path) {
+      if (!query.path || !mountModel) {
         throw error;
       }
       // control groups will throw a 403 permission denied error. If this happens return the mountModel
```

Comment:
Returning the `mountModel` was resulting in errors in the serializer if the request to fetch the mount model failed.

![image](https://user-images.githubusercontent.com/24611656/230443743-7be956b7-ee38-4e5d-ade0-eac2f60f134c.png)